app/script/assets/AssetRemoteData.js

```diff
@@ -120,7 +127,17 @@ z.assets.AssetRemoteData = class AssetRemoteData {
   }
 
   _load_buffer() {
-    return z.util.load_url_buffer(this.generate_url(), (xhr) => {
+    let generated_url;
+    try {
```

Comment:
Complicated approach. Where else are we using the `generate_url` functions? Can we turn the whole thing into the promise pattern or is this used synchroneously somewhere? Even if it is, we can turn it into a Promise chain for a cleaner approach.